### DISCUSSION TITLE
Modelling multiple routes of administration simultaneously

### DISCUSSION THREAD

**Discussion by akosuaagyeman** [nlmixr2/nlmixr2#185] [Q&A] [answered]
Hello,

Please, I am trying to model IV bolus and subcutaneous routes simultaneously following nlmixr tutorial published in CPT for the data  structure. I would like clarify if the EVID column (example data attached) is correctly set up.

Also below is the model code.  I would like to clarify if by setting up the EVID column as in the datafile, will nlmixr identify the IV bolus dose and as such will not apply the absorption rate and bioavailability terms to the IV bolus admin route?

Many thanks in advance.

```
rm(list=ls())
#Set working directory
mod.dir <- ""
setwd(mod.dir)
#load libraries
library(nlmixr2)
#
#Read in data
dat <- read.csv("Multi_routes_dat.csv")
#
#
#Modelling
two.cmt.addprop <- function() {
  ini({
    tcl <- log(0.27)      #typical value of clearance from central compartment (mL/h/kg)
    tv  <- log(39.9)      #typical value of volume of central compartment (mL/kg)
    tq <- log(1)          #typical value of inter-compartment distribution clearance (mL/h/kg)
    tvp <- log(27.56)     #typical value of volume of peripheral compartment (mL/kg)
    tka <- log(1.5)       #typical value of absorption rate (1/day)
    tfD <- log(1)       #typical value of bioavailability 
    
    prop.err <- 0.1       #residual variability
    add.err <- 0.1        #residual variability
    
  })
  model({
    cl <- exp(tcl) #individual value of cl
    v  <- exp(tv)  #individual value of v
    q <- exp(tq)   #individual value of q
    vp <- exp(tvp) #individual value of vp
    ka <- exp(tka) #individual value of ka
    fD <- exp(tfD) #individual value of bioavailability 
    
    
    k  <- cl / v  #elimination rate constant
    k12 <- q / v  #rate from central compartment to peripheral compartment
    k21 <- q / vp #rate from peripheral compartment to central compartment
    
    d/dt(depot) = -ka*depot # dose compartment
    f(depot)=fD #bioavailability of dose compartment
    d/dt(center) = ka*depot - cl/v * center - q/v * center + q/vp * periph # central compartment 
    d/dt(periph) = q/v * center - q/vp * periph # peripheral compartment
    
    cp = center/v # concentration in plasma
    
    
    cp ~ add(add.err)+prop(prop.err) # define error model
    
  })
}
#Check model
nlmixr2(two.cmt.addprop) 
#
#Run model 
fit.two.cmt.addprop <- nlmixr2(two.cmt.addprop, dat, est = "focei",  
                            table = tableControl(cwres=TRUE, npde=TRUE))
#
#Save model 
saveRDS(fit.two.cmt.addprop, "fit.two.cmt.addprop.rds")
#
#Read model back in
fit.two.cmt.addprop <-readRDS("fit.two.cmt.addprop.rds")
#
#Print
print(fit.two.cmt.addprop)
#
#
plot(fit.two.cmt.addprop)
#
#END
```

[Multi_routes_dat.csv](https://github.com/nlmixr2/nlmixr2/files/12491542/Multi_routes_dat.csv)

**Comment by mattfidler** [comment]
This is incorrect you need to use `101` for bolus doses to the central compartment and `201` for bolus doses to the second compartment.  If you want to have an infusion/zero order release to the second compartment you would need to use 2 doses, `10201` with a positive rate for `amt` when the infusion starts and `10201` with the negative value of the same rate to turn it off.

That being said, it is often easier to use nonmem-style events discussed here:

https://nlmixr2.github.io/rxode2/articles/rxode2-event-types.html

**Reply by akosuaagyeman** [reply]
Many thanks Matt. 
I guess I will achieve the same outcome if I set EVID=1 for dose, EVID=0 for observations and use CMT column to flag the route of administration?

**Reply by billdenney** [reply, marked as answer]
@akosuaagyeman, That is what I do so that `nlmixr2` will handle remembering which event type and compartment matches correctly.

**Reply by akosuaagyeman** [reply]
Many thanks Bill! Very helpful

**Reply by pascal-schulthess** [reply]
I know, I'm a bit late to the party but could you give an example? Specifically I would also be interested in having multiple endpoints as well :)

**Reply by mattfidler** [reply]
The standard multiple endpoint example is here:

https://nlmixr2.org/articles/multiple-endpoints.html

**Reply by mattfidler** [reply]
If this doesn't answer your question, can you expand your question a bit.

**Reply by pascalschulthess** [reply]
Thanks @mattfidler, I found the multiple endpoints example as well. But could you give and example dataset of how to do IV infusion and SC dosing? Thanks

**Reply by mattfidler** [reply]
Hi @pascal-schulthess 

The `sc` and the `iv` infusion can be modeled in multiple ways.

The `sc` can be treated as a first order absorption and dosed to a "depot" department, and the IV infusion could be in the central compartment.

Another possibility is to have 2 zero order infusions based on the `sc` and `iv` infusions.  With an `iv` infusion usually the infusion times are known, and with the `sc` infusion you would use `rate = -1` or `rate = -2` and estimate the rate or the duration of the zero order `sc` infusion.

These sorts of dosing scenarios are described in the rxode2 dosing vignette here (linked specifically to infusion dosing information):

https://nlmixr2.github.io/rxode2/articles/rxode2-event-types.html#infusion-doses

**Comment by mattfidler** [comment]
Yes.  It is more common to use this approach.

**Reply by akosuaagyeman** [reply]
Many thanks Matt!
